pkg/providers/vsphere/setupuser/validator.go

```diff
@@ -0,0 +1,117 @@
+package setupuser
```

Comment:
How do we intend on someone using this?

Reply:
I'm imagining adding another file, perhaps `pkg/providers/vsphere/setupuser/setup.go`, that defines something like `SetupVSphereUser` and calls this validation logic.

The command could then call `setupuser.SetupVSphereUser(ctx, filepath, ...)`.

I have little to no commitment to most of this structure. The only goal I was trying to achieve with a new package was some level of logical separation between vsphere code oriented around cluster create/upgrade/delete and this new workflow.

What do you have in mind?